app/models/reminder.rb

```diff
@@ -1,20 +1,42 @@
 class Reminder < ActiveRecord::Base
   belongs_to :user
-  validates :user, presence: true
+  belongs_to :line
 
-  after_create :remind
+  validates :user, presence: true
+  validates :line, presence: true
+  validate :user_name_and_phone_number
+  validates :user, uniqueness: {
+    scope: :line,
+    message: 'already has a reminder for this line'
+  }
 
   def remind
     @twilio_number = ENV['TWILIO_NUMBER']
     sid = ENV['TWILIO_ACCOUNT_SID']
     token = ENV['TWILIO_AUTH_TOKEN']
     @client = Twilio::REST::Client.new(sid, token)
-    # time_str = ((self.time).localtime).strftime("%I:%M%p on %b. %d, %Y")
-    reminder = "Hi #{user.name}. Time to exercise! #{}."
+    reminder = "Hi, #{user.name}. Construction on #{line.name} line will start within an hour. Visit http://localhost:3000/lines/#{line.id}/constructions for construction details"
     message = @client.account.messages.create(
       from: @twilio_number,
       to: user.phone_number,
       body: reminder,
     )
   end
+
+  def self.deliver_due
+    time_range = Time.current.to_s(:time)..1.hour.from_now.to_s(:time)
+    joins({line: :constructions}).where(
+      constructions: {start_time: time_range}
+      ).where(
+      'start_date <= ? and end_date >= ?', Date.today, Date.today
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Indent the first parameter one step more than the start of the previous line.<br>Do not use `Date.today` without zone. Use `Time.zone.today` instead.